src/skin/legacyskinparser.cpp

```diff
@@ -143,6 +143,9 @@ LegacySkinParser::LegacySkinParser(ConfigObject<ConfigValue>* pConfig,
 }
 
 LegacySkinParser::~LegacySkinParser() {
+    foreach (ControlObject* co, m_skinObjects) {
```

Comment:
Legeacy skin parse dies after the skin widget is returned. 
We need to parent the controls to the widget.